### PR TITLE
Add provisioning based on OS

### PR DESCRIPTION
**Pull request by nathanleclaire**
Closes https://github.com/docker/machine/issues/699

Continuation of work originally PRed in https://github.com/docker/machine/pull/546

This also fixes a swarm regression in master.

cc @ehazlett @sthulb

### CODE REVIEW

**Comment by nathanleclaire** [comment]
Update: I also migrated the "config" naming scheme to "options" like we discussed.

File: libmachine/provision/utils.go [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/docker/machine/libmachine/auth"
+	"github.com/docker/machine/libmachine/provision/pkgaction"
+	"github.com/docker/machine/libmachine/swarm"
+	"github.com/docker/machine/utils"
+)
+
+type DockerConfig struct {
```
Comment (nathanleclaire) [inline libmachine/provision/utils.go]: use of `DockerConfig` and `EngineConfig` persists here

File: test/integration/helpers.bash [hunk 2]
```diff
@@ -12,6 +12,19 @@ else
     ARCH="386"
 fi
 MACHINE_BIN_NAME=docker-machine_$PLATFORM-$ARCH
+BATS_LOG=${MACHINE_ROOT}/bats.log
+
+touch ${BATS_LOG}
```
Comment (nathanleclaire) [inline test/integration/helpers.bash]: I don't have `truncate` on my Mac, so I've changed this for now - we can modify it later to something more elegant if we want.

Comment (ehazlett) [inline test/integration/helpers.bash]: Ah ok -- no problem.

**Comment by ehazlett** [comment]
Tested against several drivers, swarm etc.  LGTM

Thanks!

**Pull request merged**